crates/wasmparser/src/validator.rs

```diff
@@ -50,6 +50,8 @@ fn test_validate() {
 #[cfg(feature = "component-model")]
 mod component;
 #[cfg(feature = "component-model")]
+pub(crate) use component::{CanonicalOptions, StringEncoding};
```

Comment:
Would moving more validation within `component_types` avoid the need for this? Or is this needed for other reasons?